core/src/main/java/cz/xtf/core/waiting/Waiter.java

```diff
@@ -67,6 +69,15 @@
 	 */
 	Waiter logPoint(LogPoint logPoint);
 
+	/**
+	 * Set the level of severity for a log message.
+	 *
+	 * @param level what level of severity should be used to log the message.
+	 * @return this
+	 * @see Level
+	 */
+	Waiter level(Level level);
```

Comment:
It might be better to do a default interface impl. To prevent any breaking changes of custom waiters. I'd say there's minimal chance of custom waiters being present in the wild, but we all know the drill... :)

@Crumby wdyt?

Reply:
@dsimansk Agreed. We can't break anything by making this default similarly to `onSuccess()`.

Reply:
So

```
	default Waiter level(Level level) {
		throw new UnsupportedOperationException("Method onIteration hasn't been implemented.");
	}
```

would it be?

Reply:
Fixed